### PR TITLE
chores(deps): bump dependencies cosmos-sdk `v0.47.10` and ibc-go `v7.4.1`

### PR DESCRIPTION
**Pull request by VictorTrustyDev**
Bumps:
- Cosmos-SDK to v0.47.10
- IBC go to v7.4.1

Notes:
- `--chain-id` is now mandatory flag when start chain, eg: `ethermintd start --chain-id ethermint_9000-1`
- This PR should be Squash Merged.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,20 +3,25 @@ module github.com/evmos/ethermint
...
+	github.com/cosmos/cosmos-proto v1.0.0-beta.5
+	github.com/cosmos/cosmos-sdk v0.47.10
 	github.com/cosmos/go-bip39 v1.0.0
-	github.com/cosmos/gogoproto v1.4.7
-	github.com/cosmos/ibc-go/v6 v6.1.0
+	github.com/cosmos/gogoproto v1.4.10
+	github.com/cosmos/ibc-go/v7 v7.4.1
 	github.com/davecgh/go-spew v1.1.1
 	github.com/ethereum/go-ethereum v1.10.26
-	github.com/gogo/protobuf v1.3.3
```
Comment (spoo-bar) [inline go.mod:18]: `github.com/gogo/protobuf` is deprecated and all references should be replaced with `github.com/cosmos/gogoproto`

**Comment by spoo-bar** [comment]
Using this as the base of the Dymension Hub upgrade to sdk0.47 fails with the ethermint modules not being able to access the `ctx.ConsensusParams()`. More details of the issue here: https://github.com/cosmos/cosmos-sdk/issues/16494

A potential workaround is to pass in the ConsensusParamsKeeper to the ethermint's FeemarketKeeper.
And in the `feemarketkeeper.CalculateBaseFee`, using this 
```golang
// consParams := ctx.ConsensusParams() // this doesn't work during upgrade as value is empty
consParams, err := k.consensusKeeper.Get(ctx)
if err != nil {
	return nil
}
```
And additionally, within the same func, replace
https://github.com/dymensionxyz/ethermint/blob/148a401a09dbd0b394f7e2fac5a661a67b17e39c/x/feemarket/keeper/eip1559.go#L62
with
```go
if consParams != nil && consParams.Block != nil && consParams.Block.MaxGas > -1 {
```

**Comment by VictorTrustyDev** [comment]
@spoo-bar 
I patched the _consensus params issue_ by [[1]](https://github.com/dymensionxyz/ethermint/pull/27/commits/f776492d463507cf3ccbdd9d9e1131e7b30221c9) and [[2]](https://github.com/dymensionxyz/ethermint/pull/27/commits/f5816c3619b596986f2e0f9432d3361bcbeb626b).
I have altered your suggestion a bit, look at [[2]](https://github.com/dymensionxyz/ethermint/pull/27/commits/f5816c3619b596986f2e0f9432d3361bcbeb626b)

_gogo proto_ replacement done by [[3]](https://github.com/dymensionxyz/ethermint/pull/27/commits/2da5b9c9e59230a8d7f25c43d5eaa6a71d47e384) and [[4]](https://github.com/dymensionxyz/ethermint/pull/27/commits/2f581674fa376e9e8ffaf7cd10404247ebcfd140)

File: x/evm/types/query.pb.go [hunk 2]
```diff
@@ -9,10 +9,10 @@ import (
 	github_com_cosmos_cosmos_sdk_types "github.com/cosmos/cosmos-sdk/types"
 	query "github.com/cosmos/cosmos-sdk/types/query"
 	_ "github.com/cosmos/gogoproto/gogoproto"
-	grpc1 "github.com/gogo/protobuf/grpc"
-	proto "github.com/gogo/protobuf/proto"
+	grpc1 "github.com/cosmos/gogoproto/grpc"
+	proto "github.com/cosmos/gogoproto/proto"
+	github_com_cosmos_gogoproto_types "github.com/cosmos/gogoproto/types"
 	_ "github.com/gogo/protobuf/types"
```
Comment (VictorTrustyDev) [inline x/evm/types/query.pb.go]: @spoo-bar , about _gogo proto update_

My bad, patched some in [[3]](https://github.com/dymensionxyz/ethermint/pull/27/commits/2da5b9c9e59230a8d7f25c43d5eaa6a71d47e384).
There is one place I failed to replace is this one, I have no idea why it still appears even tho I re-gen proto multiple times.

```suggestion
```

Comment (VictorTrustyDev) [reply x/evm/types/query.pb.go]: nvm, it's gone now, by [[4]](https://github.com/dymensionxyz/ethermint/pull/27/commits/2f581674fa376e9e8ffaf7cd10404247ebcfd140)

File: go.mod (hunk 1 above)
Comment (VictorTrustyDev) [reply go.mod:18]: Done, please check ty

**Comment by spoo-bar** [comment]
Looks good! Ty!

**Review by spoo-bar — APPROVED** [review]

**Review by omritoptix — APPROVED** [review]

**Pull request merged**